### ISSUE TITLE
Error(s) in loading state_dict for ResNet

### ISSUE THREAD

**Issue by barcavia** [chuangchuangtan/NPR-DeepfakeDetection#1]
Hi, 
first of all, great work!

I encountered an issue while attempting to train the model, specifically during the phase where the ResNet50 state dict is being loaded. The error message I encountered is as follows:
size mismatch for conv1.weight: copying a param with shape torch.Size([64, 3, 7, 7]) from checkpoint, the shape in current model is torch.Size([64, 3, 3, 3]).

in your ResNet class, you are using kernel_size=3 in self.conv1, but according to resnet state_dict, the kernel_size should be 7.
how did you manage to solve this when you trained your model?

Thanks in advance

**Comment by chuangchuangtan** [comment]
Do not need to load pre-trained models for NPR during training.  The classifier is modified from ResNet50.
If considering adopting resnet50 as the classifier, please replace the model with https://github.com/pytorch/vision/blob/v0.3.0/torchvision/models/resnet.py

**Comment by barcavia** [comment]
Thank you for the immediate response!
in your code, you are using pretrained=True while training (from Trainer
class):
self.model = resnet50(pretrained=True)
so it loads the pretrained resnet50.
so from your resonse, if I understand correctly, when you trained your
model, you set pretrained=False?
the weights of the model were randomly initiallized?

**Comment by chuangchuangtan** [comment]
Yes, the model is randomly initialized. We will correct this mistake. Thank you for your time.

**Comment by barcavia** [comment]
Hi,
after running the corrected code, I wasn't able to replicate your
evaluation results.
here is what I get:

(0 progan    ) acc: 99.9; ap: 100.0
(1 stylegan  ) acc: 93.2; ap: 99.8
(2 stylegan2 ) acc: 93.5; ap: 99.7
(3 biggan    ) acc: 69.7; ap: 82.1
(4 cyclegan  ) acc: 68.7; ap: 91.8
(5 stargan   ) acc: 57.3; ap: 73.5
(6 gaugan    ) acc: 64.4; ap: 82.3
(7 deepfake  ) acc: 57.5; ap: 76.6
*(8 Mean      ) acc: 75.5; ap: 88.2*

while you reported:
*mean acc: 92.5, mean ap: 96.1*

**Comment by chuangchuangtan** [comment]
If possible, please show more training hyperparameters and information about the training process.

**Comment by barcavia** [comment]
hi,
I run your code exactly how you instructed in your repository:

python train.py --name 4class-resnet-car-cat-chair-horse --dataroot
{CNNDetection-Path} --classes car,cat,chair,horse --batch_size 32
--delr_freq 10 --lr 0.0002 --niter 50

**Comment by barcavia** [comment]
What were your training hyperparameters and information about the training
process?

**Comment by chuangchuangtan** [comment]
I'm sorry for the late reply. Recently, I have been on the Lunar New Year holiday. Soon, I will release more detailed training parameters.

**Comment by chuangchuangtan** [comment]
We have added seed settings and made some modifications to details. In addition, we have released the [conda environment] (https://drive.google.com/file/d/16MK7KnPebBZx5yeN6jqJ49k7VWbEYQPr/view?usp=sharing), [training log on Serve A](https://drive.google.com/drive/folders/1X2q7mwss6Yj4NjO9oz_U2_Rk4OUDnw9N?usp=sharing), and [training log on Server B](https://drive.google.com/drive/folders/1PHFiOiOznpPcjsRSSMMIJLhyaFsBWAKD?usp=sharing). 




Our GPU Server A information is as follows:(Paper results)
CPU           : Hygon C86 7169 24-core Processor
GPU           : A4000
Nvidia Driver : 495.29.05
Ubuntu Version: 20.04.1 LTS

Please train the network model using the following command.
'''
CUDA_VISIBLE_DEVICES=3 python  train.py --name 4class-resnet_car_cat_chair_horse_bs32_ --dataroot {Path of CNNDetection} --classes car,cat,chair,horse --batch_size 32 --delr_freq 10 --lr 0.0002 --niter 51
'''

Our GPU Server B information is as follows:
CPU           : Intel(R) Xeon(R) Gold 6248R
GPU           : 3090
Nvidia Driver : 470.74
Ubuntu Version: 20.04.1 LTS

Please train the network model using the following command.
'''
CUDA_VISIBLE_DEVICES=3 python  train.py --name 4class-resnet_car_cat_chair_horse_bs32_ --dataroot {Path of CNNDetection} --classes car,cat,chair,horse --batch_size 32 --delr_freq 10 --lr 0.0002 --niter 98
'''
